### PR TITLE
#72 end to end pipeline tutorial

### PR DESCRIPTION
**Pull request by JVonderwell**
Tutorial 2: Complete Pipeline

### CODE REVIEW

**Review by gmichalo — APPROVED** [review]

File: tutorial/Tutorial_2.ipynb [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+   "metadata": {},
+   "source": [
+    "### Step 2: Error Detection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before we are able to make repairs, we need to decide what makes a cell \"clean\". This is where error detection comes in. We split the cells into two categories, \"clean\" and \"don't-know\". Using that split, our later steps of the process will be able to learn the features of a \"clean\" cell and perform inference on the values of the \"don't-know\" cells. It is simply a type of weak supervision."
```
Comment (thodrek) [inline tutorial/Tutorial_2.ipynb]: We need to be a bit more clear here. Elements from the following statements have to appear. "HoloClean is a supervised error repairing system. In contrast to traditional supervision, we do not ask users to label individual data cell but rely on more high-level supervision signals."  These signals are DCs or other error detectors and we basically require the don't know clean split. Also please add a pointer to the blog post.

**Review by gmichalo — COMMENTED** [review]
On the part where it is written : 
"Denial Constraints are the driving force behind this process. Each denial constraint can be thought of as a feature. The weights of these features are determined..."

I think we should mention the other signals that we have (init , cooccur)

File: tutorial/Tutorial_2.ipynb [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+   "metadata": {},
+   "source": [
+    "With the \"clean\" and \"don't-know\" split defined, we are ready to perform repairs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Denial Constraints are the driving force behind this process. Each denial constraint can be thought of as a feature. The weights of these features are determined by training a logistic regression model on the clean data cells via a neural network. Once those weights are defined, we use this model to perform inference on the \"don't-know\" cells and insert the most likely value for each cell."
```
Comment (thodrek) [inline tutorial/Tutorial_2.ipynb]: The statement "The weights of these features are determined by training a logistic regression model on the clean data cells via a neural network." is somewhat imprecise. Yes logistic regression is a one level neural net but having the two together will mess up folks. You can say "Denial constraints are used as features in a softmax regression model. The clean cells are used as training examples to learn the parameters (weights) of this model"

File: tutorial/Tutorial_2.ipynb [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial will simply use the default parameters for our neural network. Customization of parameters like learning rate, batch size, and number of epochs is described in Tutorial 4 and is recommended for performance-critical applications."
```
Comment (thodrek) [inline tutorial/Tutorial_2.ipynb]: Make sure you are consistent and either use neural network or softmax model. I am fine with either. Just be consistent.

Comment (thodrek) [reply tutorial/Tutorial_2.ipynb]: Overall comment: Add hyperlinks to other tutorials.

**Review by thodrek — CHANGES_REQUESTED** [review]

**Review by j48zheng — COMMENTED** [review]
I think we should pushed a version where it fully ran. The output currently stops at the training step.

**Review by gmichalo — APPROVED** [review]

**Review by thodrek — DISMISSED** [review]
Only one additional change:
For the blogpost the link doesn't show properly. Also switch the link to this http://pages.cs.wisc.edu/~thodrek/blog/holoclean.html

**Review by gmichalo — APPROVED** [review]
